### PR TITLE
🌱 envtest: use Kubernetes 1.23.1

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
Looks like 1.23.1 is the latest available version. Given that we're testing against 1.23.1 in our e2e tests I 
think we should do the same in our unit/integration tests.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -39,7 +39,7 @@ export GO111MODULE=on
 #
 # Kubebuilder.
 #
-export KUBEBUILDER_ENVTEST_KUBERNETES_VERSION ?= 1.22.0
+export KUBEBUILDER_ENVTEST_KUBERNETES_VERSION ?= 1.23.1
```
Comment (fabriziopandini) [inline Makefile:42]: Q: should we document in https://cluster-api.sigs.k8s.io/reference/versions.html that this is the Kuberenets version we are testing management clusters with, as well as the minimum Kubernetes version we run tests against?

Comment (sbueringer) [reply Makefile:42]: Sounds reasonable. I'll take a closer look at the doc tomorrow and update accordingly.

Comment (sbueringer) [reply Makefile:42]: Rolled back as discussed.

**Review by fabriziopandini — COMMENTED** [review]
Thanks!
/lgtm
/approve

**Pull request merged**